lsp-mode.el

```diff
@@ -5103,10 +5106,24 @@ It will show up only if current point has signature help."
       (funcall action-handler action)
     (lsp--send-execute-command command arguments?)))
 
-(lsp-defun lsp-execute-code-action ((action &as &CodeAction :command? :edit?))
+(defun lsp-execute-code-action (action)
   "Execute code action ACTION.
-If ACTION is not set it will be selected from `lsp-code-actions-at-point'."
+If ACTION is not set it will be selected from `lsp-code-actions-at-point'.
+Request codeAction/resolve for more info if server supports."
   (interactive (list (lsp--select-action (lsp-code-actions-at-point))))
+  (if (-> (lsp--server-capabilities)
+          (lsp:server-capabilities-code-action-provider?)
+          (lsp:code-action-options-resolve-provider?))
+      (lsp-request-async
```

Comment:
I would expect that to be sync. Imagine that you perform the code action and you switch the buffer - the code action won't be performed.

Reply:
I thought this is expected since we wouldn't block user input, but I'll change it